src/Layout/Page/Page.component.tsx

```diff
@@ -2,50 +2,215 @@
-        flex: 0 0 auto;
-        max-width: 100vw;
-    }
+    max-width: 100%;
+`;
 
-    footer {
-        flex: 0 0 auto;
-        max-width: 100vw;
-    }
+const StyledLayout = styled.section<StyledLayoutProps>`
+    max-width: ${props => props.layoutWidth};
+    background-color: ${props => props.layoutBackgroundColor};
+    margin: 0 auto;
 `;
 
-const StyledBodyElement = styled.section`
-    flex: 1 0 auto;
-    background-color: ${colors.silver.base};
-    max-width: 100vw;
-    align-self: stretch;
-    padding: ${sizes.padding.lg} ${sizes.padding.xl};
+const StyledContent = styled.div<StyledContentProps>`
+    max-width: ${props => props.contentWidth};
+    background-color: ${props => props.contentBackgroundColor};
+    margin: 0 auto;
 `;
 
+// Interfaces
+// ------------------------------------------------------------------------------------------------------------------
 interface PageProps {
+    /** The header of the website. This can be a component. */
+    header?: any;
+    /** The footer of the website. This can be a component. */
+    footer?: any;
+    /** The height of the header. It will not hide on overflow. */
+    headerHeight?: string | number;
+    /** The height of the footer. It will not hide on overflow. */
+    footerHeight?: string | number;
+    /** Additional classname. */
+    className?: string;
+    children?: any;
+}
+
+interface DefaultLayoutProps {
+    /** The width of the layout, edge to edge. */
+    layoutWidth?: string | number;
+    /** The width of the content within the layout. */
+    contentWidth?: string | number;
+    /** The background color of the layout. Default is transparent. */
+    layoutBackgroundColor?: string;
+    /** The background color of the content within the layout area. Default is transparent. */
+    contentBackgroundColor?: string;
+    /** Additional classname. */
+    className?: string;
+    children?: any;
+}
+
+interface SidebarLayoutProps {
+    /** The width of the layout, edge to edge. */
+    layoutWidth?: string | number;
+    /** The width of the content within the layout. */
+    contentWidth?: string | number;
+    /** The width of the area for the sidebar. */
+    sidebarWidth?: string | number;
+    /** If the sidebar is on the left or the right of the layout */
+    sidebarAlign?: SidebarAlignment;
+    /** The actual sidebar. This can be a component. */
+    sidebar?: any;
+    /** The background color of the layout. Default is transparent. */
+    layoutBackgroundColor?: string;
+    /** The background color of the content within the layout area. Default is transparent. */
+    contentBackgroundColor?: string;
+    /** Additional classname. */
     className?: string;
     children?: any;
 }
 
-export const Page = ({ children }: PageProps): React.ReactElement<any> => (
-    <StyledPageElement className="anchor-page">
+interface StyledLayoutProps {
+    layoutWidth?: string | number;
+    layoutBackgroundColor?: string;
+}
+
+interface StyledContentProps {
```

Comment:
Can we break all of these components apart?

Reply:
That's my plan. I'm going to do it in an upcoming push.